plugins/woocommerce-blocks/assets/js/blocks-registry/product-collection/register-product-collection.tsx

```diff
@@ -64,9 +70,27 @@ const registerProductCollection = ( {
 		);
 	}
 
-	registerBlockVariation( blockJson.name, {
+	const isActive = (
+		blockAttrs: BlockAttributes,
+		variationAttributes: BlockAttributes
+	) => {
+		return blockAttrs.collection === variationAttributes.collection;
+	};
+
+	registerBlockVariation( BLOCK_NAME, {
 		...blockVariationArgs,
+		attributes: {
+			...DEFAULT_ATTRIBUTES,
+			...blockVariationArgs.attributes,
+			query: {
+				...DEFAULT_ATTRIBUTES.query,
+				...blockVariationArgs.attributes?.query,
+			},
+			collection: blockVariationArgs.name,
+			inherit: blockVariationArgs.attributes?.inherit || false,
```

Comment:
`inherit` is part of `query`, not `attributes` and should be moved there.

Reply:
Let's decide based on the output [here](https://github.com/woocommerce/woocommerce/pull/48141#discussion_r1633077630). If we want collections to be able to set "inherit" then I can make this change 🤝